src/js/media-time-display.js

```diff
@@ -157,31 +149,85 @@ class MediaTimeDisplay extends MediaTextDisplay {
     this.tabIndex = -1;
   }
 
+  // Own props
+
+  /**
+   * Whether to show the remaining time
+   * @type {boolean}
+   */
+  get remaining() {
+    return this.hasAttribute(Attributes.REMAINING);
+  }
+
+  set remaining(show) {
+    // don't set unless needed, could trigger an attr change event
+    if (show === this.remaining) return;
+    this.toggleAttribute(Attributes.REMAINING, show);
+  }
+
+  /**
+   * Whether to show the duration
+   * @type {boolean}
+   */
+  get showDuration() {
+    return this.hasAttribute(Attributes.SHOW_DURATION);
+  }
+
+  set showDuration(show) {
+    // don't set unless needed, could trigger an attr change event
+    if (show === this.showDuration) return;
+    this.toggleAttribute(Attributes.SHOW_DURATION, show);
+  }
+
+  // Props derived from media UI attributes
+
+  /**
+   * Get the duration
+   * @type {number | undefined} In seconds
+   */
   get mediaDuration() {
     const attrVal = this.getAttribute(MediaUIAttributes.MEDIA_DURATION);
     return attrVal != null ? +attrVal : undefined;
   }
 
+  set mediaDuration(time) {
+    this.setAttribute(MediaUIAttributes.MEDIA_DURATION, time.toString(10));
```

Comment:
what happens if time is undefined or a string value here?

it would be good I think if string numbers would still work and don't break the functionality

Reply:
Good call. It seems strings should still work (I tried `'10'.toString(10)` and it gave me back the string `'10'`), but worth addressing `time` being undefined.

Probably removing the attribute when a falsey value is provided?

Reply:
I'm thinking of accepting undefined for all the new prop setters instead of just the ones that explicitly have undefined as a return type on the getter (what I was doing currently), I think that makes sense?